client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx

```diff
@@ -125,6 +219,14 @@ const SiteSetupList = ( {
 		}
 	}, [ currentTaskId, dispatch, tasks ] );
 
+	// If specified then automatically complete the current task when viewed.
+	useEffect( () => {
+		if ( currentTask?.completeOnView ) {
+			setUserSelectedTask( true );
```

Comment:
Set to selected to ensure that it will be made (forced) visible even though it has been completed. This concept already exists with the component.

Reply:
I am not sure that `setUserSelectedTask` is appropriate here. It seems that it may be used for tracking user actions on tasks, in which case it would return wrong results since the user hasn't actually selected this task.
Also, I tried commenting it out and cannot see any visual difference (after creating a new site).

I may be missing something on the above though...

Reply:
I see why this is confusing. I think it's a naming problem.

>  It seems that it may be used for tracking user actions on tasks

It isn't...yet. But it could be. All it does is allow tasks to be "selected" (and viewed) by the User even if they are complete.

> Also, I tried commenting it out and cannot see any visual difference (after creating a new site).

Try commenting it out and then create a new site from scratch. You will land on the My Home page and the "Site created" item will be complete, but it will not be selected. Therefore you will miss the celebratory step of seeing the task content. 

Uncomment it and it will work as expected.

I think I'll rename the state and that will make the intent clearer.